packages/dds/tree/src/feature-libraries/flex-map-tree/mapTreeNode.ts

```diff
@@ -482,9 +482,31 @@ class EagerMapTreeSequenceField<T extends FlexAllowedTypes>
 	extends EagerMapTreeField<T>
 	implements FlexTreeSequenceField<T>
 {
-	public get editor(): SequenceFieldEditBuilder<ExclusiveMapTree[]> {
-		throw unsupportedUsageError("Editing an array");
-	}
+	public editor: SequenceFieldEditBuilder<ExclusiveMapTree[]> = {
+		insert: (index, newContent) => {
+			for (let i = 0; i < newContent.length; i++) {
+				const c = newContent[i];
+				assert(c !== undefined, "Unexpected sparse array content");
+				nodeCache.get(c)?.adoptBy(this, index + i);
```

Comment:
Whether or not there is a cache hit here is equivalent to whether or not this node is hydrated? And if this node were to be hydrated in the future, presumably caches would be correctly updated with the correct parentage and everything? Just making sure I'm understanding what's going on here.

Reply:
Correct - if the node is hydrated, then you'll get a cache hit, otherwise not. For example, in the following unhydrated array, there is an unhydrated node for the first element but not the second element.

`const fooArray = new FooArray([new Foo({ value: 3 }), { value: 3 }]);`

If/when inserted into the tree, both elements will have hydrated LazyFlexNodes created for them. In the case of the first element, that means swapping out the MapTreeNode for the LazyFlexNode, and in the case of the second element, that just means creating a LazyFlexNode. 

The MapTree for the second element cannot become part of a different unhydrated node later, because the user does not have a handle to the MapTree node.